client/lib/feed-lookup/index.js

```diff
@@ -0,0 +1,75 @@
+// External Dependencies
+var isEmpty = require( 'lodash/isEmpty' ),
+	head = require( 'lodash/head' ),
+	lruCache = require( 'lru-cache' );
+
+// Internal Dependencies
+var inflight = require( 'lib/inflight' ),
+	wpcom = require( 'lib/wp' );
+
+var FeedLookupCache,
+	FeedLookup;
+
+const cache = lruCache( 10 );
+
+FeedLookupCache = {
+	get: function( url ) {
+		return cache.get( url );
+	},
+
+	set: function( url, id ) {
+		cache.set( url, id );
+	}
+};
+
+function requestKey( feedId ) {
+	return `feed-lookup-${feedId}`;
+}
+
+function discover( feedUrl ) {
+	const key = requestKey( feedUrl );
+
+	if ( inflight.requestInflight( key ) ) {
```

Comment:
I'd like to return the Promise that will fulfil when the inflight request completes here instead of `undefined`